Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -304,6 +330,185 @@ public boolean onPreferenceClick(Preference preference) {
+                FragmentActivity activity = getActivity();
+                if (activity == null) {
+                    return;
+                }
+
+                int colorResource = ThemeUtils.isLightTheme(activity) ?
+                        R.color.red_50 :
+                        R.color.red_20;
+                int colorRed = ContextCompat.getColor(activity.getApplicationContext(),
+                        colorResource);
+                dialogDeleteAccount
+                        .getButton(AlertDialog.BUTTON_POSITIVE)
+                        .setTextColor(colorRed);
+            }
+        });
+        dialogDeleteAccount.show();
+    }
+
+    private Callback getAccountDeleteCallbackHandler() {
+        return new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                FragmentActivity activity = getActivity();
+                if (activity == null) {
+                    return;
+                }
+
+                AppLog.add(Type.ACCOUNT, "Failure while calling server to delete account");
+
+                activity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        closeProgressDialogDeleteAccount();
+                        showDialogDeleteAccountError();
+                    }
+                });
+            }
+
+            @Override
+            public void onResponse(Call call, final Response response) throws IOException {
+                FragmentActivity activity = getActivity();
+                if (activity == null) {
+                    return;
+                }
+
+                activity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        closeProgressDialogDeleteAccount();
+
+                        // The delete account requests return 200 when the request was processed
+                        // successfully. These requests send an email to the user with instructions
+                        // to delete the account. This email is valid for 24h. If the user sends
+                        // another request for deletion and the previous request is still valid,
+                        // the server sends a response with code 202. We take both 200 and 202 as
+                        // successful responses. Both codes are handled by isSuccessful()
+                        if (response.isSuccessful()) {
+                            AppLog.add(Type.ACCOUNT, "Request to delete account was successful");
+
+                            showDeleteAccountConfirmationDialog();
+                        } else {
+                            AppLog.add(Type.ACCOUNT, "Request to delete account had an " +
+                                    "error. Error code: " + response.code());
+
+                            showDialogDeleteAccountError();
+                        }
```

Comment:
just a thought, I don't think it's a good idea to expose OkHttp's CallBack here, it's making the fragment responsible for network logic (response.isSuccessFull).
I would prefer to have some kind of interface as listener for the account deletion, with two simple methods: `onSuccess` and `onFailure`, which we can then pass to `makeDeleteAccountRequest`, and in `AccountNetworkUtils` we would use the `Callback` to handle the logic of success and failure and forward it to our listener.

Reply:
HI @hichamboushaba 👋 . Yes, I agree, it is not the responsibility of the fragment. I have added a handler in 28c1446a4d80a59936bc5de194bd53560cc0a070 and move the fragment to a weak reference in the handler (dd8199e4f88400f2d43936ddc55bfbeb5f1d887c) to avoid memory leaks.